File: java/7- Sorting/selectionSort.java
import java.util.Arrays;

public class selectionSort {
    public static int getMaxIndex(int[] arr, int s, int e) {
        int max = s;

        for (int i = s; i <= e; i++) {
            if (arr[max] < arr[i]) {
                max = i;
            }
        }

        return max;
    }

    public static void swap(int[] arr, int s, int e) {
        int temp = arr[s];
        arr[s] = arr[e];
        arr[e] = temp;

    }

    public static void main(String[] args) {
        int[] arr = { 4, 5, 1, 2, 3 };

        for (int i = 0; i < arr.length; i++) {
            int last = arr.length - i - 1;
            int maxIndex = getMaxIndex(arr, 0, last);

            swap(arr, maxIndex, last);
        }

        System.out.println(Arrays.toString(arr));
    }
}

File: Learning Recursion/SelectionSort.py
myArr = [4,3,2,1]

def SelectionSort(arr,r,c,maxIn):
    if r == 0:
        return
    
    if c < r:
        if arr[c] > arr[maxIn]:
            SelectionSort(arr,r,c+1,c)
        else:
            SelectionSort(arr,r,c+1,maxIn)
    else:
        temp = arr[maxIn]
        arr[maxIn] = arr[r-1]
        arr[r-1]=temp

        SelectionSort(arr,r-1,0,0)

SelectionSort(myArr,len(myArr),0,0)
print(myArr)